python/cudf/cudf/core/dataframe.py

```diff
@@ -1456,6 +1456,98 @@ def add(self, other, axis="columns", level=None, fill_value=None):
 
         return self._apply_op("add", other, fill_value)
 
+    def update(
+        self,
+        other,
+        join="left",
+        overwrite=True,
+        filter_func=None,
+        errors="ignore",
+    ):
+        """
+        Modify a DataFrame in place using non-NA values from another DataFrame.
+
+        Aligns on indices. There is no return value.
+
+        Parameters
+        ----------
+        other : DataFrame, or object coercible into a DataFrame
+            Should have at least one matching index/column label with the
+            original DataFrame. If a Series is passed, its name attribute must
+            be set, and that will be used as the column name to align with the
+            original DataFrame.
+
+        join : {`left`}, default `left`
+            Only left join is implemented, keeping the index and
+            columns of the original object.
+
+        overwrite : {True, False}, default True
+            How to handle non-NA values for overlapping keys:
+            True: overwrite original DataFrame’s values with values from other.
+            False: only update values that are NA in the original DataFrame.
+
+        filter_func : None
+            filter_func is not supported yet
+            Return True for values that should be updated.
+
+        errors : {‘raise’, ‘ignore’}, default ‘ignore’
+            If ‘raise’, will raise a ValueError if the DataFrame and other
+            both contain non-NA data in the same place.
+
+
+        Returns
+        -------
+        None : method directly changes calling object
+
+        Raises
+        -------
+        ValueError
+            - When ``errors``= 'raise' and there’s overlapping non-NA data.
```

Comment:
Need space to correctly render code highlighting.

```suggestion
            - When ``errors`` = 'raise' and there's overlapping non-NA data.
```